packages/composer-json-manipulator/tests/ValueObject/ComposerJsonTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\ComposerJsonManipulator\Tests\ValueObject;
+
+use Generator;
+use PHPUnit\Framework\TestCase;
+use Symplify\ComposerJsonManipulator\ValueObject\ComposerJson;
+use Symplify\ComposerJsonManipulator\ValueObject\ComposerJsonSection;
+
+final class ComposerJsonTest extends TestCase
+{
+    public function testJsonArray() : void
+    {
+        $composerJson = new ComposerJson();
+        $composerJson->setRepositories(['type' => 'vcs']);
+        $composerJson->setRequireDev(['dev-dependency']);
+        $composerJson->setRequire(['dependency']);
+        $composerJson->setConflicting(['conflicting']);
+        $composerJson->setPreferStable(true);
+        $composerJson->setMinimumStability('dev');
+        $composerJson->setAutoload(['autoload']);
+        $composerJson->setAutoloadDev(['dev-autoload']);
+        $composerJson->setReplace(['replace']);
+        $composerJson->setConfig(['config-key' => 'value']);
+        $composerJson->setExtra(['extra-key' => 'value']);
+        $composerJson->setName('package-name');
+        $composerJson->setDescription('My description');
+        $composerJson->setLicense('MIT');
+        $composerJson->setScripts(['scripts']);
+        $composerJson->setAuthors(['name' => 'ruud']);
+        $composerJson->setType('library');
+
+        $array = $composerJson->getJsonArray();
+        $this->assertSame(['type' => 'vcs'], $array[ComposerJsonSection::REPOSITORIES]);
+        $this->assertSame(['dev-dependency'], $array[ComposerJsonSection::REQUIRE_DEV]);
+        $this->assertSame(['dependency'], $array[ComposerJsonSection::REQUIRE]);
+        $this->assertSame(['conflicting'], $array[ComposerJsonSection::CONFLICTING]);
+        $this->assertSame(true, $array[ComposerJsonSection::PREFER_STABLE]);
+        $this->assertSame('dev', $array[ComposerJsonSection::MINIMUM_STABILITY]);
+        $this->assertSame(['autoload'], $array[ComposerJsonSection::AUTOLOAD]);
+        $this->assertSame(['dev-autoload'], $array[ComposerJsonSection::AUTOLOAD_DEV]);
+        $this->assertSame(['replace'], $array[ComposerJsonSection::REPLACE]);
+        $this->assertSame(['config-key' => 'value'], $array[ComposerJsonSection::CONFIG]);
+        $this->assertSame(['extra-key' => 'value'], $array[ComposerJsonSection::EXTRA]);
+        $this->assertSame('package-name', $array[ComposerJsonSection::NAME]);
+        $this->assertSame('My description', $array[ComposerJsonSection::DESCRIPTION]);
+        $this->assertSame('MIT', $array[ComposerJsonSection::LICENSE]);
+        $this->assertSame(['scripts'], $array[ComposerJsonSection::SCRIPTS]);
+        $this->assertSame('library', $array[ComposerJsonSection::TYPE]);
+        $this->assertSame(['name' => 'ruud'], $array[ComposerJsonSection::AUTHORS]);
+        }
+}
```

Comment:
This is very hard to read and probably to maintain.
I though this issue was about missed merge. 

How can it we tested with less work and better readability?

Reply:
I don't know, how can it be tested that it becomes easier to read? 

Right now, there is no test and it does not work as expected. This change makes sure that it does work with a test to guard it. 

Sure, it can probably be done with some kind of Reflection trick but does that make it easier to understand?

Reply:
I imagine the real use case as merge 2 `composer.json` and one `composer.json` as output. That's where the issue was spot, right?

Such test would be easy to read and maintain.

Reply:
I think the test to verify that `authors` are covered should be inside the `composer-json-manipulator` package, not within the `monorepo-builder` package. 

I'll see if I can create a functional test in the `composer-json-manipulator` package.

Reply:
There is no need to do that, as it's in a monorepo both test suites are run at once.

We save lot of maintenance by testing the higher element, in monorepo-builder, and other packages as well.

Reply:
I just pushed a new test, before I saw your reply. What do you think?

Reply:
Not sure why everything is failing now, I just rebased with master. 🤔

Reply:
Looks like "replace" in composer.json needs update to current version